### PR TITLE
[Windows] Implement new directories hierarchy

### PR DESCRIPTION
**Pull request by vpolikarpov-akvelon**
# Description

This PR implements new unified directories hierarchy for Windows image source code:

- `assets`: Configuration files that are copied to the image during the build process.
  - `post-gen`: Scripts and files related to post-image generation tasks.
- `scripts`:
  - `build`: Scripts that are executed during the image build process.
  - `helpers`: Scripts used by other scripts during the image build process.
  - `docs-gen`: Script and PowerShell modules for generating software and configuration reports.
  - `tests`: Test scripts and related files for validating the image.
- `templates`: Packer templates for the image build process.
- `toolsets`: JSON files that define the software versions to be installed on the image.

> ⚠️
> Packer templates are moved to `templates` directory and renamed.
> Pipelines that reference them will need an update

#### Related issue: https://github.com/actions/runner-images-internal/issues/5513

CI Runs: [Windows 2019](https://github.com/actions/runner-images-ci/actions/runs/6588504782), [Windows 2022](https://github.com/actions/runner-images-ci/actions/runs/6588490307)

## Check list
- [x] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [x] Documentation is updated (if applicable)
- [x] Changes are tested and related VM images are successfully generated

### CODE REVIEW

**Comment by shamil-mubarakshin** [comment]
`windows2019.yml` and `windows202.yml` labeled trigger needs update as well

File: images/windows/scripts/build/Update-ImageData.ps1 [hunk 1]
```diff
@@ -1,45 +1,45 @@
...
+$imageVersion = $env:IMAGE_VERSION
+$imageVersionComponents = $imageVersion.Split('.')
+$imageMajorVersion = $imageVersionComponents[0]
+$imageMinorVersion = $imageVersionComponents[1]
+$imageDataFile = $env:IMAGEDATA_FILE
+$githubUrl="https://github.com/actions/runner-images/blob"
+
+if (Test-IsWin22) {
+    $imageLabel = "windows-2022"
+    $softwareUrl = "${githubUrl}/win22/$imageMajorVersion.$imageMinorVersion/images/win/Windows2022-Readme.md"
```
Comment (shamil-mubarakshin) [inline images/windows/scripts/build/Update-ImageData.ps1:15]: should we update paths here as well

File: docs/create-image-and-azure-resources.md [hunk 2]
```diff
@@ -269,9 +269,9 @@ Generated tool versions and details can be found in related projects:
 
 > :warning: These scripts are intended to run on a VM deployed in Azure
 
-The user, created during the image generation, does not exist in the result image hence some configuration files related to the user's home directory need to be changed as well as the file permissions for some directories. Scripts for that are located in the `post-generation` folder in the repository:
+The user, created during the image generation, does not exist in the result image hence some configuration files related to the user's home directory need to be changed as well as the file permissions for some directories. Scripts for that are located in the `post-gen` folder in the repository:
 
-- Windows: <https://github.com/actions/runner-images/tree/main/images/win/post-generation>
+- Windows: <https://github.com/actions/runner-images/tree/main/images/windows/post-gen>
```
Comment (shamil-mubarakshin) [inline docs/create-image-and-azure-resources.md:274]: assets/post-gen

**Review by shamil-mubarakshin — APPROVED** [review]

File: images/windows/scripts/build/Configure-DynamicPort.ps1
Comment (Alexey-Ayupov) [inline images/windows/scripts/build/Configure-DynamicPort.ps1:1]: According to style guide: "Each file should have a newline at the end;"

Comment (vpolikarpov-akvelon) [reply images/windows/scripts/build/Configure-DynamicPort.ps1:1]: This PR only updates directory structure. Refactoring of script themselves will be performed later via separate request.

**Review by mikhailkoliada — APPROVED** [review]

**Pull request merged**